lib/src/widgets/screen_select_dialog.dart

```diff
@@ -42,38 +110,43 @@ class ScreenSelectDialog extends Dialog {
 
   void _ok(BuildContext context) {
     _timer?.cancel();
-    for (var item in _subscriptions) {
-      item.cancel();
+    for (var element in _subscriptions) {
+      element.cancel();
     }
     Navigator.pop<rtc.DesktopCapturerSource>(context, _selectedSource);
   }
 
   void _cancel(BuildContext context) {
     _timer?.cancel();
-    for (var item in _subscriptions) {
-      item.cancel();
+    for (var element in _subscriptions) {
+      element.cancel();
     }
     Navigator.pop<rtc.DesktopCapturerSource>(context, null);
   }
 
   Future<void> _getSources() async {
     try {
       var sources = await rtc.desktopCapturer.getSources(types: [_sourceType]);
-      for (var item in sources) {
-        logger.info('name: ${item.name}, id: ${item.id}, type: ${item.type}');
-      }
-      _stateSetter?.call(() {
-        for (var item in sources) {
-          _sources[item.id] = item;
+      for (var element in sources) {
+        if (kDebugMode) {
+          print(
+              'name: ${element.name}, id: ${element.id}, type: ${element.type}');
         }
-      });
+      }
       _timer?.cancel();
-      _timer = Timer.periodic(const Duration(seconds: 2), (timer) {
+      _timer = Timer.periodic(const Duration(seconds: 3), (timer) {
```

Comment:
one concern is with this running on slower computers, if it takes more than 3s to update complete a set of thumbnails, it could just be doing this constantly?

Reply:
Here timer actually only refresh the source list, add or remove, the thumbnail is done in another thread, only the content of the window is changed (image changed), and the thumbnail change event will be emitted.
https://github.com/webrtc-sdk/webrtc/pull/37/files#diff-2ed6dd2f4d3d1562a1c0810e03c7ae6f504a4472009708ed62a538d2997893ebR157

Reply:
ah ok, understood. thank you for the explanation.